src/realm/replication.hpp

```diff
@@ -302,13 +306,16 @@ class Replication : public _impl::TransactLogConvenientEncoder, protected _impl:
     /// remains in the Realm file, there are practical complications, and for
     /// that reason, such switching shall not be supported.
     ///
-    /// The \ref hist_Sync history type can only be used if the stored history
-    /// type is also \ref hist_Sync, or when there is no top array
-    /// yet. Additionally, when the stored history type is \ref hist_Sync, then
-    /// all subsequent sesssions must have the same type. These restrictions
-    /// apply because such a history needs to be maintained persistently across
-    /// sessions. That is, the contents of such a history is not obsolete at the
-    /// end of the session, and is in general needed during subsequent sessions.
+    /// The \ref hist_SyncClient history type can only be used if the stored
+    /// history type is also \ref hist_SyncClient, or when there is no top array
+    /// yet. Liewise, he \ref hist_SyncServer history type can only be used if
+    /// the stored history type is also \ref hist_SyncServer, or when there is
+    /// no top array yet. Additionally, when the stored history type is \ref
+    /// hist_SyncClient or \ref hist_SyncServer, then all subsequent sessions
+    /// must have the same type. These restrictions apply because such a history
+    /// needs to be maintained persistently across sessions. That is, the
+    /// contents of such a history is not obsolete at the end of the session,
+    /// and is in general needed during subsequent sessions.
```

Comment:
Can this whole explanation be simplified? What I get out of this is: 
If you have no previous history, you can start any kind of history. Once you have started one type, you have to stick with that. At least if we say that OutOfRealm history is now obsolete.
@kspangsege @finnschiermer